src/v/cloud_storage/tests/cloud_storage_e2e_test.cc

```diff
@@ -249,7 +249,7 @@ TEST_P(EndToEndFixture, TestProduceConsumeFromCloudWithSpillover) {
         ASSERT_TRUE(archiver.sync_for_tests().get());
         archiver
           .upload_next_candidates(
-            archival::archival_stm_fence{.unsafe_add = true})
+            archival::archival_stm_fence{.emit_rw_fence_cmd = false})
```

Comment:
consider in the future fixing/editing previous commits rather than just adding new commits fixing other commits in the same PR

this tool is very helpful to achieve this: https://github.com/tummychow/git-absorb

Reply:
I'm using this tool and it's often messes up commit history like this. It only works well if different changes target different places in the code. 
Anyway, this is a response to the code review request so IMO it's fine to add it as a separate commit.

Reply:
The problem is that next reviewer who reviews commit by commit again don't see these changes until the very end. Harder to review.

Reply:
this is just a rename